src/Microsoft.PowerShell.Commands.Utility/commands/utility/sort-object.cs

```diff
@@ -239,13 +245,12 @@ protected override void EndProcessing()
             // Track the number of items that will be output from the data once it is sorted
             int sortedItemCount = dataToProcess.Count;
 
-            // If -Top & -Bottom were not used, or if -Top or -Bottom would return all objects, invoke
-            // an in-place full sort
-            if ((Top == 0 && Bottom == 0) || Top >= dataToProcess.Count || Bottom >= dataToProcess.Count)
+            // If -Stable, -Top & -Bottom were not used, invoke an in-place full sort
+            if (!Stable && Top == 0 && Bottom == 0)
             {
                 sortedItemCount = FullSort(dataToProcess, comparer);
             }
-            // Otherwise, use an indexed min-/max-heap to perform an in-place sort of all objects
+            // Otherwise, use an indexed min-/max-heap to perform an in-place, stable sort of all objects
```

Comment:
Please add to comment that Heap sort performs a stable sort, and provide short description of stable sort.

Reply:
Done and coming in a commit shortly.